index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const Joi = require('joi');
+const emailer = require('./email');
+
+// Joi Schema Validation
+const SCHEMA_ADDRESS = Joi.string().email();
+const SCHEMA_ADDRESSES = Joi.array()
+    .items(SCHEMA_ADDRESS)
+    .min(0);
+const SCHEMA_STATUS = Joi.string();
+const SCHEMA_STATUSES = Joi.array()
+    .items(SCHEMA_STATUS)
+    .min(0);
+const SCHEMA_BUILD_SETTINGS = Joi.alternatives().try(
+    Joi.object().keys({ addresses: SCHEMA_ADDRESSES, statuses: SCHEMA_STATUSES }),
+    Joi.string().email()
+    );
+const SCHEMA_BUILD_DATA = Joi.object()
+    .keys({
+        settings: SCHEMA_BUILD_SETTINGS,
+        status: SCHEMA_STATUS
+    });
+const SCHEMA_SMTP_CONFIG = Joi.object()
+    .keys({
+        host: Joi.string(),
+        port: Joi.number().integer(),
+        from: Joi.string().email()
+    });
+
+const DESCRIPTION_MAP = {
+    SUCCESS: 'Everything looks good!',
+    FAILURE: 'Did not work as expected.',
+    ABORTED: 'Aborted mid-flight',
+    RUNNING: 'Testing your code...',
+    QUEUED: 'Looking for a place to park...'
+};
+const DEFAULT_STATUSES = ['FAILURE'];
+
+let config;
```

Comment:
I would suggest keeping these with the `EmailNotifier` class and treating them like member field.